gluten-core/src/main/scala/io/glutenproject/execution/BatchScanExecTransformer.scala

```diff
@@ -60,9 +60,15 @@ class BatchScanExecTransformer(
     case _ => new StructType()
   }
 
-  override def getInputFilePaths: Seq[String] = scan match {
-    case fileScan: FileScan => fileScan.fileIndex.inputFiles.toSeq
-    case _ => Seq.empty
+  override def getInputFilePaths: Seq[String] = {
+    if (BackendsApiManager.isVeloxBackend) {
```

Comment:
Suggest to do a separation for different backends's by introducing a new interface. See BackendSettingsApi. Ditto for the below hive table scan transformer.

Reply:
This solution is more general for parsing partition column info compared with extract from file path, I dont think introduce a new interface for it in BackendSettingsApi is a good idea, BackendSettingsApi already has lots of settings, we should control it to avoid settings inflate more and more.

I add a comments for BasicScanExecTransformer#getInputFilePaths, it's an expensive interface and should be removed in the future, in our tests, do not invoke getInputFilePaths in driver main thread explicitly could gain some performance boost, Spark also let it be lazy.

Reply:
Could you file an issue on removing this method in the future? The fix can be pending till it is also applicable to CH. Thanks!

Reply:
> Could you file an issue on removing this method in the future?

Sure.

> The fix can be pending till it is also applicable to CH.

You mean pending current PR? I dont really understand the reason, could you give more thoughts on this?

Reply:
I meant the fix for removing this method can be pending. Sorry for my ambiguous comment.

Reply:
Already add new issue for that. https://github.com/oap-project/gluten/issues/2472